crates/iota-framework/packages/iota-system/tests/rewards_distribution_tests.move

```diff
@@ -972,11 +974,137 @@ module iota_system::rewards_distribution_tests {
         scenario_val.end();
     }
 
+    #[test]
+    fun test_constant_exchange_rates_with_no_rewards() {
+        set_up_iota_system_state();
+        let mut scenario_val = test_scenario::begin(VALIDATOR_ADDR_1);
+        let scenario = &mut scenario_val;
+
+        // Add different stake to different pools
+        stake_with(STAKER_ADDR_1, VALIDATOR_ADDR_1, 1, scenario);
+        stake_with(STAKER_ADDR_2, VALIDATOR_ADDR_2, 50, scenario);
+        stake_with(STAKER_ADDR_3, VALIDATOR_ADDR_3, 3, scenario);
+
+        // need to advance epoch so validator's staking starts counting
+        advance_epoch(scenario);
+
+        // Advance a couple of epochs to make the exchange rate different than 1.
+        // Validator 1: 10% commission.
+        set_commission_rate_and_advance_epoch(VALIDATOR_ADDR_1, 1000, scenario);
+        advance_epoch_with_reward_amounts(0, 1, scenario);
+
+        // Stake and unstake different amounts to one of the pools
+        stake_with(STAKER_ADDR_4, VALIDATOR_ADDR_3, 150, scenario);
+        unstake(STAKER_ADDR_3, 0, scenario);
+
+        // Advance another epoch without rewards
```

Comment:
You advance the epoch with reward = 1 right after this comment.

Reply:
nice catch, line 1001 shouldn't be there. I was actually surprised that the test passed without accounting for fixed point math for validator 3, but that's why...

Reply:
done